src/Compilers/Core/Portable/MetadataReference/CachedAssemblySymbolList.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal sealed class CachedAssemblySymbolList
+    {
+        private readonly object _guard = new object();
+        private readonly WeakList<IAssemblySymbol> _cachedSymbols = new WeakList<IAssemblySymbol>();
+
+        internal void ForEach<T>(Action<T> action)
```

Comment:
>void ForEach<T>(Action<T> action) [](start = 17, length = 33)

Consider merging this with the method below since the callers are just creating a list in each case: `internal List<T> Copy<T>(Func<T, bool> predicate)`.